File: examples/tests/issue173.cc
#include "ortools/base/commandlineflags.h"
#include "ortools/base/logging.h"
#include "ortools/linear_solver/linear_solver.h"

namespace operations_research {
void SolveLP() {
  MPSolver solver("test", MPSolver::CBC_MIXED_INTEGER_PROGRAMMING);
  const double kInfinity = solver.infinity();
  MPVariable* const x = solver.MakeNumVar(-kInfinity, kInfinity, "x");

  MPObjective* const objective = solver.MutableObjective();
  objective->SetMaximization();
  objective->SetCoefficient(x, 1);

  MPConstraint* const constraint = solver.MakeRowConstraint(0, 5);
  constraint->SetCoefficient(x, 1);

  solver.Solve();
}

void BreakLoop() {
  for (int i = 0; i < 500; i++) {
    SolveLP();
  }
}
}  // namespace operations_research

int main(int argc, char** argv) {
  absl::ParseCommandLine(argc, argv);
  operations_research::BreakLoop();
  return 0;
}

File: examples/contrib/Issue173.java
package com.google.ortools.contrib;

import com.google.ortools.Loader;
import com.google.ortools.linearsolver.MPConstraint;
import com.google.ortools.linearsolver.MPObjective;
import com.google.ortools.linearsolver.MPSolver;
import com.google.ortools.linearsolver.MPVariable;

public class Issue173 {
  public static void breakit() {
    for (int i = 0; i < 50000; i++) {
      solveLP();
    }
  }

  private static void solveLP() {
    MPSolver solver = MPSolver.createSolver("CBC");
    if (solver == null) {
      System.out.println("Could not create solver CBC");
      return;
    }
    MPVariable x = solver.makeNumVar(Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, "x");

    final MPObjective objective = solver.objective();
    objective.setMaximization();
    objective.setCoefficient(x, 1);

    MPConstraint constraint = solver.makeConstraint(0, 5);
    constraint.setCoefficient(x, 1);

    solver.solve();
  }

  public static void main(String[] args) throws Exception {
    Loader.loadNativeLibraries();
    breakit();
  }
}
